pkg/controller/knicluster/knicluster_controller.go

```diff
@@ -276,11 +325,32 @@ func (r *ReconcileKNICluster) Reconcile(request reconcile.Request) (reconcile.Re
 	} {
 		err = f(instance, reqLogger)
 		if err != nil {
+			status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
+				Type:    kniv1alpha1.ConditionDegraded,
+				Status:  corev1.ConditionTrue,
+				Reason:  "ReconcileFailed",
+				Message: fmt.Sprintf("Failed reconcilitation %v", err),
+			})
+
+			_ = r.client.Status().Update(context.TODO(), instance)
 			return reconcile.Result{}, err
 		}
 	}
 
-	return reconcile.Result{}, nil
+	status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
```

Comment:
Probably need to dig deeper before marking `kni-operator` as "Available" but for now this illustrates the point.